default.conf

```diff
@@ -0,0 +1,13 @@
+server {
+    listen       9002 ssl;
+    ssl_certificate /var/serving-cert/tls.crt;
+    ssl_certificate_key /var/serving-cert/tls.key;
```

Comment:
Did we test this Dockerfile and NGINX config on an OCP cluster? Just want to make sure if this assumption that's on ODF still holds.

Reply:
it should be true IMO... plugin needs serving certificate for https... in odf plugin, ODF operator adds it to the deployment...
in our case noobaa deployer should be doing that...

Reply:
Ah okay, so this PR is blocked by [[DFMS-50] Adding TLS support to prometheus instance](https://issues.redhat.com/browse/DFMS-50).

Reply:
nope... not blocked by it... DFMS-50 is to put external Prometheus behind https...
@v-harihar is working on creating a plugin deployment from the deployer, there in the operator CSV we need to specify the mountPath...

Reply:
> in our case noobaa deployer should be doing that...

- At the moment the deployer does not have this functionality.
- Vinayak's PR targets `drf-console` UI pod's reconciliation from the deployer, not non-self-signed cert creation, https://github.com/red-hat-storage/mcg-osd-deployer/pull/16/files#diff-416223883b53a9d25c101e5a5c44f8861bc7c18cbeed9ce7ee4fe0ee0dd37c24R48.
- When the TLS functionality is in place to enable HTTPS for the external Prometheus endpoint, it will essentially use RedHat's cert-serving service (operator) to generate and fetch the necessary key and certificate, and the path where they are stored will be mounted to the UI pod, which this PR is dependent on, https://github.com/red-hat-storage/ocs-osd-deployer/pull/149/files#diff-1d409d0b85193d4fa78899757bfc7c41ce41a06da0fb1f48d01d4cf7187ce654R517.

PLMK if that makes sense.

Reply:
When u will create a Service for the dfr-plugin, you need to add an annotation to it... once that Service is created it will take care of creating the serving certificates automatically as well... we need to mount that Secret to the plugin Deployment as well. At least this is what is happening with ODF right now.
Maybe same certificate can be used for Prometheus or vice versa ??

Reply:
Okay, if that's the case, then I'm assuming that this is blocked by Vinayak's PR?

Reply:
> then I'm assuming that this is still blocked by Vinayak's PR?

I think `/var/serving-cert`, I  mean this PR should be merged 1st to verify that from my side